oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/OAuth2InvalidTokenError.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.core;
+
+/**
+ * A representation of an OAuth 2.0 Invalid Token Error.
+ *
+ * <p>
+ * At a minimum, an error response will contain an error code. The error code will always
+ * be "invalid_token"
+ *
+ * @author Jonas Erbe
+ * @since 5.0
+ * @see OAuth2ErrorCodes
+ */
+public class OAuth2InvalidTokenError extends OAuth2Error {
```

Comment:
Personally I favor stronger typing over redundance. The new error class makes returning the wrong error impossible since the compiler fails. But you are right, the public API can't be changed, so I will remove the class.